enterprise/app/services/enterprise/message_templates/hook_execution_service.rb

```diff
@@ -10,6 +10,9 @@ def trigger_templates
   end
 
   def should_process_captain_response?
+    # we don't check the captain usage limits here intentionally
+    # as the usage limits are checked in the job itself
+    # this is to ensure that the handoff can be done there if it's required
```

Comment:
We should the check for responses here IMO and keep the response builder simple. The conversation won't be in a pending state as there is no `active_bot?`. So running the job to process handoff is unnecessary.